pkg/util/throughput/throughput.go

```diff
@@ -0,0 +1,58 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package throughput
+
+import (
+	"sync"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/util/ewma"
+)
+
+var (
+	watcher           sync.Once
+	throughputWatcher *throughputCalc
+)
+
+// InitWatcher initializes the watcher that watches the throughput of incoming data.
+func InitWatcher(calculateEvery time.Duration) {
+	if calculateEvery == 0 {
+		// Watching throughput is disabled.
+		return
+	}
+	watcher.Do(func() {
+		throughputWatcher = NewCal(calculateEvery)
+		go throughputWatcher.run()
+	})
+}
+
+type throughputCalc struct {
+	every time.Duration
+	cal   *ewma.Rate
+}
+
+func NewCal(every time.Duration) *throughputCalc {
```

Comment:
does not need to be public
```suggestion
func newThroughputCal(every time.Duration) *throughputCalc {
```